### PR TITLE
Feature: SiteNow v3 WYSIWYG and media

### LINKED ISSUE #1989: Move to collegiate responsive image styles for default profile

### LINKED ISSUE #1990: Review and consolidate input formats for default, v2, and collegiate

### LINKED ISSUE #2060: Allow minimal_html and filtered_html to have an <hr> with a class
# Problem 

With the removal of the spacer/separator block, we need the ability to add a spacing and clear option within the markup: https://github.com/uiowa/uids/issues/289

# Proposed solution 

Allow `minimal_html` and `filtered_html` to use the `element--spacer-separator` class on `<hr>` tags.   Ideally, we could open up the `<hr>` class for future classes since I could see us developing different line styles.

### LINKED ISSUE #2099: Adopt Core's Media Library for V3 default
## Problem

One of the main differences between default v3 and the collegiate split is how media is handled within the WYSIWYG. Launching v3 with v2's media/wywisyg implementation increases our technical debt for when we finally do approach https://github.com/uiowa/uiowa/issues/865 and presents technical challenges in developing for two different solutions where everything else is pretty much aligned.

## Proposed Solution

- [x] Split off the entity browser media implementation to the v2 split.
- [x] Move collegiate configuration surrounding the media library implementation into v3 default config.
- [x] Investigate only allowing certain view modes for certain media types - https://github.com/uiowa/uiowa/issues/865#issuecomment-682194469 https://www.drupal.org/project/drupal/issues/3097416
- [x] Update uids_base to handle the styling around each media type
- [x] Update media_entity_instagram
- [x] Update media_entity_facebook as soon as media library support is added, otherwise update documentation to indicate that new Facebook items can only be added through /media/add and not through the WYSIWYG. _(not needed for MVP)_
  - https://www.drupal.org/project/media_entity_facebook/issues/3120085
- [x] https://github.com/uiowa/uiowa/issues/1989
- [x] https://github.com/uiowa/uiowa/issues/1990

### PR DESCRIPTION
**Pull request by pyrello**
Resolves #2099 
Relates to #1867
Resolves #2060
Resolves #1990 
Resolves #1989 

To test the Instagram oEmbed you need config overrides local for the Facebook App API it uses. This is included on all remote environments.

- [x] Add credentials to Acquia's secret file in each environment/app.

# To Test  
- `composer install`

## Config changes
- Test syncing 1 or 2 existing v2 sites (e.g. `blt ds --site=default`)
- Test syncing sites with site splits
- Test syncing 1 or 2 collegiate sites

## File Deletion

- Find a really old file like the default section image. Remove it from the section and delete it from media. 0 Usage.
- In file_managed table, you should see the status of that file as 0. Copy the value for created and paste it as the value for changed. Otherwise you have to wait for the 3 month timer. Run cron. File should be removed from file_managed, file_usage and from the local file directory.
- This only works for new deletions, not old orphaned files.

### CODE REVIEW

**Comment by pyrello** [comment]
@joewhitsitt Do you have an example of what needs to be added to local configuration for Facebook oEmbed?

**Comment by joewhitsitt** [comment]
So far, configuration seems to have been moved appropriately.

v2 plain:

So far, not seeing any differences in WYSIWYG/Media handling compared to PROD. I was able to add each of the media types and the renders spit out something.

v3 plain: 

- Facebook is missing an add option in the v3 WYSIWYG, but that support needs to come from contrib so, /media/add for new Facebook entities in the meantime.
- should only see the responsive image view modes for images. oEmbeds should have the original/sm/md/lg options though I don't think there are differences beyond remote video in how those are displayed.

v2/v3:

- Instagram add/render should work if you have the same credentials listed in uiowa.dev's secret file locally.
- Twitter wysiwyg editor representation is lacking and is not really identifiable (out of scope). https://github.com/uiowa/uiowa/issues/2147

Haven't tested other custom splits yet.

**Comment by cory-skeers** [comment]
Been working on a default V3 install, and images and remote video oEmbeds are working well, with the appropriate view modes available.

Is theming the media select widget in scope for this, and/or was the plan to use the media_library_theme_reset module?

**Comment by pyrello** [comment]
@cory-skeers I would consider it out of scope for this, mostly since we are sprinting towards a beta version of V3. Good idea for a follow-up issue.

That being said, if you have some ideas for sprucing it up, feel free to add them to the PR. We can always touch it up more in a follow-up.

**Comment by pyrello** [comment]
Here is some more config I think can be moved from the default profile to the V2 split:

**Media view modes**
* Edgy
* SiteNow 16:9

**Responsive image styles**
* Edgy No Crop
* SiteNow 16:9

**Image styles**
* Edgy No Crop *
* Large (480x480)
* Medium (220x220)
* SiteNow 16:9 *

I think the following image styles can just be removed entirely because I can't find any reference to their machine name in the context of other config or in code:
* SiteNow Article
* SiteNow Card
* SiteNow Square Medium

**Comment by joewhitsitt** [comment]
I think I agree with everything but small, medium, large. They were provided by core and are being used for remote video on V3. With the view mode patch, if configured correctly, you won't see them for Images.

**Comment by pyrello** [comment]
@joewhitsitt I don't know if you saw that I removed those from the list of view modes to move. However I am thinking that I'll recreate those view modes so they don't have machine names of `image_small`, since we are not using them for images :)

**Comment by joewhitsitt** [comment]
Sorry, I was going email by email through notifications this AM

File: config/default/filter.format.filtered_html.yml [hunk 1]
```diff
@@ -63,7 +63,11 @@ filters:
     status: true
     weight: -49
     settings:
+<<<<<<< HEAD
```
Comment (joewhitsitt) [inline config/default/filter.format.filtered_html.yml]: These are remnants from a merge which will break on config import.

**Comment by pyrello** [comment]
@joewhitsitt or @cory-skeers Any issue with adding this file to the default profile? https://github.com/uiowa/uiowa/blob/feature_sn3_media/config/features/collegiate/pathauto.pattern.media.yml

**Comment by pyrello** [comment]
I'm trying to think of any reason why it would be necessary to maintain the separate entity view display for the page teaser in the collegiate split: https://github.com/uiowa/uiowa/blob/feature_sn3_media/config/features/collegiate/core.entity_view_display.node.page.teaser.yml

I can't think of any instance where we actually use the page teaser at this point. @mark-bennett-uiowa or @bspeare Can you think of any instances where this is used? Or can we remove this and start using the version in default? This is the part that's different: https://github.com/uiowa/uiowa/blob/dfca93e3a225a351b867cbb400f0f3ff53541d2c/config/default/core.entity_view_display.node.page.teaser.yml#L24-L32

It looks like the image field displays vs. not displaying.

**Comment by joewhitsitt** [comment]
> @joewhitsitt or @cory-skeers Any issue with adding this file to the default profile? https://github.com/uiowa/uiowa/blob/feature_sn3_media/config/features/collegiate/pathauto.pattern.media.yml

No concern from me. @cory-skeers would this path affect sitenow_migrate d7_file?

**Comment by cory-skeers** [comment]
> > @joewhitsitt or @cory-skeers Any issue with adding this file to the default profile? https://github.com/uiowa/uiowa/blob/feature_sn3_media/config/features/collegiate/pathauto.pattern.media.yml
> 
> No concern from me. @cory-skeers would this path affect sitenow_migrate d7_file?

It might need to be updated, but it should be a minor update if anything. No concerns here.

**Comment by mark-bennett-uiowa** [comment]
> I'm trying to think of any reason why it would be necessary to maintain the separate entity view display for the page teaser in the collegiate split: https://github.com/uiowa/uiowa/blob/feature_sn3_media/config/features/collegiate/core.entity_view_display.node.page.teaser.yml
> 
> I can't think of any instance where we actually use the page teaser at this point. @mark-bennett-uiowa or @bspeare Can you think of any instances where this is used? Or can we remove this and start using the version in default? This is the part that's different:

@pyrello I cant think of a reason either. Isnt this file one of the files that is one of the problems in the policy.clas site split also? If we can go with the default, I say go for it.

**Comment by joewhitsitt** [comment]
Latest changes to the view modes are affecting remote video selections in V3. It saves with a small view mode, but renders default and when I open the edit form again, it says original as the selection.

**Comment by pyrello** [comment]
Thanks @joewhitsitt. I'll take a look. I'm still reviewing v2 at the moment.

**Comment by pyrello** [comment]
At this point, I think v2 is all good from a config standpoint. It may still need some feature testing. I'm moving on to reviewing v3 for config. Then collegiate.

**Comment by joewhitsitt** [comment]
Tested a v2 default site and hr and things seem to be okay.

Looking at the github file diffs, some of the collegiate to sitenow_v2 "moves" have me scratching my head a bit, but it must be how github shows the diff. Lesson learned not to combine big moves and changes in the same PR..

**Comment by joewhitsitt** [comment]
I revisited Cory's regression fix commit and copied out the same media attributes as uiowa.edu prod. This solved my problem here: https://github.com/uiowa/uiowa/pull/2110#issuecomment-689742742

Added classes for the new view modes to handle size styling of remote video.

**Comment by pyrello** [comment]
@joewhitsitt You also solved another issue that had been puzzling me for the last hour or so. I thought maybe the patch had introduced a regression. Whew. Thanks!

**Comment by pyrello** [comment]
I have tested this against v2 and v3. I'm now running the sync against uiowa.edu, coronavirus.uiowa.edu, and policy.clas.uiowa.edu just to check that I don't get any config errors. I'm definitely ready for others to test, but I think this is good to go.

**Comment by pyrello** [comment]
@mark-bennett-uiowa This PR moves most config out of collegiate and into the default split. It would be helpful if you were able to test against one or two of your sites to make sure nothing is broken. I have already made sure that syncing against policy.clas.uiowa.edu works.

**Comment by joewhitsitt** [comment]
Agreed that I think this is good from a v2/v3 perspective, I think what has been done is a good step forward and would enable us to start provisioning v3 sites. Biased approval.

**Comment by cory-skeers** [comment]
Spot checked a few V2 sites I've worked on before, and everything looks good 👍

**Review by richardbporter — APPROVED** [review]
I synced a few sites and added some media locally. Since I'm not up to speed with everything changing here, I could hold this up for awhile and that seems unnecessary given others have tested/approved already.

**Comment by joewhitsitt** [comment]
I believe I added creds to all remote environments

**Comment by mark-bennett-uiowa** [comment]
@pyrello @joewhitsitt Checked this on three on my larger CLAS sites. Zero issues with config that I could see during local creation or during testing. Tested out adding media (image and remote video) on several pages and then worked great (not available in Minimal HTML) each time. Already merged, but approved. Nice work guys!

**Comment by joewhitsitt** [comment]
I think the plan is to continue testing in stage. I will be testing the instagram api credentials there.

**Pull request merged**